js/render/draw_circle.js

```diff
@@ -34,7 +34,6 @@ function drawCircles(painter, source, layer, coords) {
         var program = painter.useProgram('circle', bucket.getProgramMacros('circle', layer));
 
         gl.uniform1f(program.u_blur, Math.max(layer.paint['circle-blur'], antialias));
-        gl.uniform1f(program.u_size, layer.paint['circle-radius']);
```

Comment:
The use of `circle-radius` [here](https://github.com/mapbox/mapbox-gl-js/blob/5e3cd2796bee21cca3f3c8670a821e900a2e9ca8/js/render/draw_circle.js#L23) also needs to be removed since it might not be a constant value. This will probably require some shader changes.